src/bootstrap/config.rs

```diff
@@ -101,6 +101,43 @@ impl Display for DebuginfoLevel {
     }
 }
 
+#[derive(Default, Clone)]
+pub struct LLVMConfig {
```

Comment:
this is essentially the same as `config::Llvm` struct, right? can we reuse that instead of adding a new struct?

(btw ty for putting this in a separate commit, it made things a lot easier to review :heart:)

Reply:
The goal of this struct is to have a "ground truth" representation of all the options that can affect the build of LLVM, so that we can compare the CI version and the local version in an easy way - by just comparing the value of a struct, without manually comparing a subset of its keys.

However, there are some options under the `[llvm]` key in `config.toml` which should not be compared - notably `download-ci-llvm`. We definitely do not want to include this key when comparing the CI and local configs. So this is not 1:1 with the `LLVM` struct.

Note: I only removed `from_ci` from this struct in a later commit, to uphold what I wrote above, which is probably confusing.